Firmware/MotorControl/commands.c

```diff
@@ -335,21 +344,35 @@ void cmd_parse_thread(void const * argument) {
                     }
                 }
             }
-            // When we reach here, we are out of immediate characters to fetch out of UART buffer
-            // Now we check if there is any USB processing to do: we wait for up to 1 ms,
-            // before going back to checking UART again.
-            int USB_check_timeout = 1;
-            // Wait for signalling from USB interrupt (OTG_FS_IRQHandler)
-            osStatus semaphore_status = osSemaphoreWait(sem_usb_irq, USB_check_timeout);
-            if (semaphore_status == osOK) {
-                // We have a new incoming USB transmission: handle it
-                HAL_PCD_IRQHandler(&hpcd_USB_OTG_FS);
-                // Let the irq (OTG_FS_IRQHandler) fire again.
-                HAL_NVIC_EnableIRQ(OTG_FS_IRQn);
+            // Check if there is USB processing to do.
+            osStatus sem_stat = osSemaphoreWait(sem_usb_rx, 0);
```

Comment:
This shouldn't be 0 wait, or else this thread will never sleep if there is nothing to do. You should make the timeout 1ms again, and put back the comment:

// When we reach here, we are out of immediate characters to fetch out of UART buffer
// Now we check if there is any USB processing to do: we wait for up to 1 ms,
// before going back to checking UART again.